Xamarin.Forms.Platform.Android/Forms.cs

```diff
@@ -76,9 +77,26 @@ public static void Init(Context activity, Bundle bundle, Assembly resourceAssemb
 			SetupInit(activity, resourceAssembly);
 		}
 
+		/// <summary>
+		/// Sets title bar visibility programmatically. Must be called after Xamarin.Forms.Forms.Init() method
+		/// </summary>
+		/// <param name="visibility">Title bar visibility enum</param>
 		public static void SetTitleBarVisibility(AndroidTitleBarVisibility visibility)
 		{
+			if((Activity)Context == null) throw new NullReferenceException("Must be called after Xamarin.Forms.Forms.Init() method");
+
 			TitleBarVisibility = visibility;
+
+			if (TitleBarVisibility == AndroidTitleBarVisibility.Never)
+			{
+				if(!((Activity)Context).Window.Attributes.Flags.HasFlag(WindowManagerFlags.Fullscreen))
+					((Activity)Context).Window.AddFlags(WindowManagerFlags.Fullscreen);
```

Comment:
I've searched the net a bit. Looks like there is no way to adjust resize in full screen mode and this is by design. In that case, the developer should be aware of the consequences and design accordingly. There are hacky solutions such as toggling fullscreen on and off on keyboard display or wrapping the content in a ScrollView.